tests/php/modules/shortcodes/test-class.instagram.php

```diff
@@ -372,4 +390,136 @@ public function test_shortcodes_instagram_non_amp( $shortcode_content ) {
 		add_filter( 'jetpack_is_amp_request', '__return_false' );
 		$this->assertNotContains( 'amp-instagram', do_shortcode( $shortcode_content ) );
 	}
+
+	/**
+	 * Test the build of a set of allowed parameters from a variety of inputs.
+	 *
+	 * @dataProvider get_instagram_parameters
+	 * @covers ::jetpack_instagram_get_allowed_parameters
+	 *
+	 * @param string $url      URL of the content to be embedded.
+	 * @param array  $atts     Shortcode attributes.
+	 * @param array  $expected Array of expected parameters.
+	 *
+	 * @since 9.1.0
+	 */
+	public function test_shortcodes_instagram_allowed_parameters( $url, $atts, $expected ) {
+		$GLOBALS['content_width'] = self::CONTENT_WIDTH;
+
+		$actual = jetpack_instagram_get_allowed_parameters( $url, $atts );
+		$this->assertEquals( $expected, $actual );
+	}
+
+	/**
+	 * Variety of parameters available from an embed.
+	 *
+	 * @covers ::jetpack_instagram_get_allowed_parameters
+	 *
+	 * @since 9.1.0
+	 */
+	public function get_instagram_parameters() {
+		$base_instagram_url = 'https://www.instagram.com/p/BnMOk_FFsxg';
+
+		return array(
+			'no_query_strings_no_atts'     => array(
+				$base_instagram_url,
+				array(),
+				array(
+					'url'         => $base_instagram_url,
+					'width'       => self::CONTENT_WIDTH,
+					'height'      => '',
+					'hidecaption' => false,
+				),
+			),
+			'invalid_query_string_no_atts' => array(
+				$base_instagram_url . '?utm_source=ig_web_copy_link',
+				array(),
+				array(
+					'url'         => $base_instagram_url,
+					'width'       => self::CONTENT_WIDTH,
+					'height'      => '',
+					'hidecaption' => false,
+				),
+			),
+			'invalid_query_string_hidecaption_string_no_atts' => array(
+				$base_instagram_url . '?utm_source=ig_web_copy_link&hidecaption=true',
+				array(),
+				array(
+					'url'         => $base_instagram_url,
+					'width'       => self::CONTENT_WIDTH,
+					'height'      => '',
+					'hidecaption' => 'true',
+				),
+			),
+			'hidecaption_string_no_atts'   => array(
+				$base_instagram_url . '?hidecaption=true',
+				array(),
+				array(
+					'url'         => $base_instagram_url,
+					'width'       => self::CONTENT_WIDTH,
+					'height'      => '',
+					'hidecaption' => 'true',
+				),
+			),
+			'hidecaption_att'              => array(
+				$base_instagram_url,
+				array(
+					'hidecaption' => 'true',
+				),
+				array(
+					'url'         => $base_instagram_url,
+					'width'       => self::CONTENT_WIDTH,
+					'height'      => '',
+					'hidecaption' => 'true',
+				),
+			),
+			'invalid_atts_in_url_att'      => array(
+				$base_instagram_url,
```

Comment:
I note this test might pass if it uses the URL passed here to the one on line 479. If you specify a mismatching URL here, that should cover that possibility.

Reply:
Good point. I added 042fe543eebcea753b08bde241ade6107808a578 to try to take this into account.